Make ghcid no longer works with multi-package setup

I wonder if there's anything we can do about this. Currently ghcid will only watch and recompile if you are editing anything under `src`. GHCi is notoriously difficult to get working with multi-package setups, and I haven't found a configuration that works yet. Adding `purescript-ast` and `purescript-cst` to the ghci command results in conflicts (expecting things to be from one of the libs, but ghci has put it all in one scope).

> I tried using it before splitting the packages and ran into some error with ghcid. I had assumed that make rule was bit rotted. Was it working for you recently? If so, how'd you get it to work?

Yes, it's how I do most of my work on the compiler. It's always just worked for me.

If you add purescript-cst and purescript-ast to the command:
```
ghcid --command "stack ghci purescript:lib purescript:test:tests purescript-ast:lib purescript-cst:lib --ghci-options -fno-code"
```

You'll get errors like:
```
1: error:
    Could not load module ‘Language.PureScript.Names’
    It is a member of the hidden package ‘purescript-ast-0.1.0.0’.
    You can run ‘:set -package purescript-ast’ to expose it.
    (Note: this unloads all the modules in the current scope.)
   |
16 | import qualified "purescript-ast" Language.PureScript.Names as N
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

If you follow this advice:
```
ghcid --command "stack ghci purescript:lib purescript:test:tests purescript-ast:lib purescript-cst:lib --ghci-options '-fno-code -package purescript-ast'"
```

You'll get errors like:
```
purescript/lib/purescript-cst/.stack-work/dist/x86_64-osx/Cabal-2.4.0.
1/build/Language/PureScript/CST/Parser.hs:1500:41: error:
    • Couldn't match type ‘Language.PureScript.Names.ModuleName’
                     with ‘N.ModuleName’
      NB: ‘N.ModuleName’ is defined at
            /Users/nathan/Sites/git/purescript/lib/purescript-ast/src/Language/PureScript/Name
s.hs:(155,1)-(156,35)
          ‘Language.PureScript.Names.ModuleName’
            is defined in ‘Language.PureScript.Names’
                in package ‘purescript-ast-0.1.0.0’
      Expected type: Name N.ModuleName
        Actual type: Name Language.PureScript.Names.ModuleName
    • In the first argument of ‘happyIn26’, namely ‘r’
      In the first argument of ‘happyReturn’, namely ‘(happyIn26 r)’
      In the expression: happyReturn (happyIn26 r)
     |
1500 |         ) (\r -> happyReturn (happyIn26 r))
     |                                         ^
```
But this happens for _everything_ referenced from purescript-ast (so it's not just a cst parser issue). I'm not sure why "purescript-ast" is hidden, and I'm guessing that using the `-package` flag conflicts with package imports in ghci.

--- a/Makefile
+++ b/Makefile
@@ -7,10 +7,10 @@ help: ## Print documentation
 	@grep -E '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}'
 
 ghcid: ## Run ghcid to quickly reload code on save.
-	ghcid --command "stack ghci purescript:lib purescript:test:tests --ghci-options -fno-code"
+	ghcid --command "stack ghci purescript:exe:purs purescript:lib purescript:test:tests purescript-ast purescript-cst --main-is purescript:exe:purs --ghci-options -fno-code"
 
 ghcid-test: ## Run ghcid to quickly reload code and run tests on save.
-	ghcid --command "stack ghci purescript:lib purescript:test:tests --ghci-options -fobject-code" \
+	ghcid --command "stack ghci purescript:lib purescript:test:tests purescript-ast purescript-cst --ghci-options -fobject-code" \
 	    --test "Main.main"
 
 build: ## Build the package.

